docs/source/developer_documentation.rst

```diff
@@ -26,20 +27,31 @@ A variety of coding conventions are enforced by automated tools in continuous in
 PyRenew Principles
 ------------------
 
--  Variable naming conventions
+Variable naming conventions
+---------------------------
+-  Use the ``data_`` prefix for (potentially) observed data.
+-  Use the ``_rv`` suffix for random variables.
+-  Use the ``observed_`` for the output of sample statements where ``obs`` is a ``data_`` prefixed object.
```

Comment:
I am not entirely sure that these prefixes make complete sense. Particularly, I make an explicit distinction between observed and unobserved data. In principle, everything can be called data, so it is superfluous to use that as a prefix. Instead, I would say `sampled_` for *any* output (deterministic or random), leave the `observed_` to actually observed data, and keep the `_rv` suffix.

Reply:
@gvegayon @dylanhmorris How about this? The `observed_` prefix is nice because it maps to the `obs` argument in `numpyro.sample`, though I understand why it is confusing say the `observed_` object may not actually be observed.
```
-  Use the ``observed_`` prefix for (potentially) observed data.
-  Use the ``_rv`` suffix for random variables.
-  Use the ``sampled_`` for the output of sample statements where ``obs`` is a ``obserced_`` prefixed object.
-  Thus, code which may reasonably written like ``infections = infections.sample(x, obs=infections)`` should instead be written ``sampled_infections = infections_rv.sample(x, obs=observed_infections)``.
```